system_server crash when ending session with secondary users

After the last system update, it happens when I end session with a secondary user to switch back to the owner.

```
type: crash

SystemUptimeMs: 49368
Process: system_server
Timestamp: 2024-12-12 10:37:49.928+0100
Build: google/shiba/shiba:15/AP4A.241205.013/2024120900:user/release-keys
Crash-Handler: com.android.internal.os.RuntimeInit$KillApplicationHandler
Loading-Progress: 1.0
Dropped-Count: 0

java.lang.RuntimeException: Error receiving broadcast Intent { act=android.intent.action.LOCALE_CHANGED flg=0x1200810 } in com.android.server.inputmethod.SystemLocaleWrapper$LocaleChangeListener@8c35e95
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0(LoadedApk.java:1827)
	at android.app.LoadedApk$ReceiverDispatcher$Args.$r8$lambda$mcNAAl1SQ4MyJPyDg8TJ2x2h0Rk(Unknown Source:0)
	at android.app.LoadedApk$ReceiverDispatcher$Args$$ExternalSyntheticLambda0.run(D8$$SyntheticClass:0)
	at android.os.Handler.handleCallback(Handler.java:991)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.os.HandlerThread.run(HandlerThread.java:85)
	at com.android.server.ServiceThread.run(ServiceThread.java:46)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.view.inputmethod.InputMethodInfo.isSystem()' on a null object reference
	at com.android.server.inputmethod.InputMethodManagerService.resetDefaultImeLocked(InputMethodManagerService.java:1298)
	at com.android.server.inputmethod.InputMethodManagerService.onActionLocaleChanged(InputMethodManagerService.java:742)
	at com.android.server.inputmethod.InputMethodManagerService$$ExternalSyntheticLambda3.onLocaleChanged(R8$$SyntheticClass:0)
	at com.android.server.inputmethod.SystemLocaleWrapper$LocaleChangeListener.onReceive(SystemLocaleWrapper.java:104)
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0(LoadedApk.java:1819)
	... 8 more
```

--- a/services/core/java/com/android/server/inputmethod/InputMethodManagerService.java
+++ b/services/core/java/com/android/server/inputmethod/InputMethodManagerService.java
@@ -1294,9 +1294,11 @@ public final class InputMethodManagerService implements IInputMethodManagerImpl.
         // Do not reset the default (current) IME when it is a 3rd-party IME
         String selectedMethodId = bindingController.getSelectedMethodId();
         final InputMethodSettings settings = InputMethodSettingsRepository.get(userId);
-        if (selectedMethodId != null
-                && !settings.getMethodMap().get(selectedMethodId).isSystem()) {
-            return;
+        if (selectedMethodId != null) {
+            InputMethodInfo selectedMethod = settings.getMethodMap().get(selectedMethodId);
+            if (selectedMethod != null && !selectedMethod.isSystem()) {
+                return;
+            }
         }
         final List<InputMethodInfo> suitableImes = InputMethodInfoUtils.getDefaultEnabledImes(
                 context, settings.getEnabledInputMethodList());

